src/map/atcommand.c

```diff
@@ -5277,18 +5277,58 @@ ACMD(follow)
 }
 
 /*==========================================
- * @dropall by [MouseJstr]
- * Drop all your possession on the ground
- *------------------------------------------*/
+* @dropall by [MouseJstr] and [Xantara]
+* Drop all your possession on the ground based on item type
+*------------------------------------------*/
 ACMD(dropall)
 {
-	for (int i = 0; i < sd->status.inventorySize; i++) {
+	int type = -1;
+	int i, count = 0;
+	struct item_data *item_data = NULL;
```

Comment:
item_data used only inside ``if``, this mean variable must be moved into ``if`` block.
``i`` variable should be inside ``for``: ``for (int i = ...``
other vars probably should be moved too

Reply:
hmm, aren't previously all variables shall move to the top? something like to avoid confusion/ease to spot the variable for other dev to may modify it in future.

anyway, updated.

Reply:
in past it was because old visual studio.
But correct way to use closer variable scope